url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/resources/DB3DProperties.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Sourcecode of the
+ *
+ * University of Osnabrueck
+ * Institute for Geoinformatics and Remote Sensing
+ *
+ * Copyright (C) Researchgroup Prof. Dr. Martin Breunig
+ *
+ */
+package de.uos.igf.db3d.resources;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * A preconfigured properties object... <br>
+ * Easy to use:<br>
+ * <code>
+ * DB3DProperties.getProperty("mypropertyname");
+ * DB3DProperties.setProperty("mypropertyname", "42");
+ * DB3DProperties.store();
+ * </code>
+ * 
+ * @author Edgar Butwilowski
+ */
+public final class DB3DProperties {
+
+	private final static Properties db3dProperties = new Properties();
+	private static File propertiesFile;
+
+	static {
+
+		/*
+		 * CAUTION!: do not use logging in this static constructor, since the
+		 * logging architecture of Db3dLogger itself depends on an object of
+		 * this class. Simply use printStackTrace() method instead.
+		 */
+
+		try {
+
+//			String appFolderString = DB3DProperties.class.getResource(
+//					DB3DProperties.class.getSimpleName() + ".class").toString();
+//			
+//			System.out.println(":::::::::::::::::::::::::::::::::: "+appFolderString);
+//
+//			/*
+//			 * cut
+//			 * "file:/.../DB3DoDB/bin/de/uos/igf/db3d/resources/Db3dProperties.class"
+//			 * to ".../DB3DoDB/":
+//			 */
+//			appFolderString = appFolderString.substring(5, appFolderString
+//					.length()
+//					- (DB3DProperties.class.getCanonicalName().length() + 10));
+//
+//			String propertiesFolderString = appFolderString + "properties";
+			
+			String propertiesFolderString = "etc/";
+			
+			File propertiesFolder = new File(propertiesFolderString);
+
+			if (!propertiesFolder.exists()) {
+				propertiesFolder.mkdir();
+			}
+
+			propertiesFile = new File(propertiesFolderString
+					+ "/db3d.properties");
+
+			if (!propertiesFile.exists()) {
+				propertiesFile.createNewFile();
+			}
+
+			if (propertiesFile.exists()) {
+				db3dProperties.load(new FileInputStream(propertiesFile));
+			}
+
+		} catch (Exception ex) {
+			// Db3dLogger can not be used here, since the logger needs the
+			// Db3dProperties, so:
+			System.out.println("DB3DProperties: Can not find db3d.properties file "
+					+ "in properties/ folder of the server.");
+		}
+
+	}
+
+	// prohibit instantiation:
+	private DB3DProperties() {
+	}
+
+	public static void store(String comments) throws IOException {
+		if (propertiesFile != null && propertiesFile.exists()) {
+			try {
+				FileOutputStream fw = new FileOutputStream(propertiesFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fw*. Execution paths that do not contain closure statements were detected. Ensure *fw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).